datacube/api/core.py

```diff
@@ -6,14 +6,15 @@
 import uuid
 import collections.abc
 from itertools import groupby
-from typing import Set, Union, Optional, Dict, Tuple, cast
+from os import PathLike
+from typing import Any, Set, Union, Optional, Dict, Tuple, cast
```

Comment:
Might as well get rid of `Union` and `Dict` imports while you're at it

Reply:
There will be another PR coming soonish with a major overhaul of typehints.

Given that, I'm not sure it's worth cluttering up this PR further.

Reply:
Cleaned up typehints in this file.  Rest of core can wait for a more widespread typehint cleanup.